src/React.AspNet/HtmlHelperExtensions.cs

```diff
@@ -144,55 +149,34 @@ private static IReactEnvironment Environment
 		/// <returns>JavaScript for all components</returns>
 		public static IHtmlString ReactInitJavaScript(this IHtmlHelper htmlHelper, bool clientOnly = false)
 		{
-			try
-			{
-				return GetScriptTag(Environment.GetInitJavaScript(clientOnly));
-			}
-			finally
+			return new ActionHtmlString(writer =>
 			{
-				Environment.ReturnEngineToPool();
-			}
+				try
+				{
+					WriteScriptTag(writer, bodyWriter => Environment.GetInitJavaScript(bodyWriter, clientOnly));
+				}
+				finally
+				{
+					Environment.ReturnEngineToPool();
+				}
+			});
 		}
 
-		private static IHtmlString GetScriptTag(string script)
+		private static void WriteScriptTag(TextWriter writer, Action<TextWriter> bodyWriter)
 		{
-#if LEGACYASPNET
-			var tag = new TagBuilder("script")
-			{
-				InnerHtml = script,
-			};
-
+			writer.Write("<script");
 			if (Environment.Configuration.ScriptNonceProvider != null)
 			{
-				tag.Attributes.Add("nonce", Environment.Configuration.ScriptNonceProvider());
+				writer.Write(" nonce=\"");
+				writer.Write(Environment.Configuration.ScriptNonceProvider());
```

Comment:
I see that we're manually constructing HTML tags in existing code.. as long as we're not passing user-generated data to these methods, this should be OK (although it would be nice to sanitize the data going into these attributes just in case)

Reply:
yea, original tag builder has a similar code https://github.com/aspnet/Mvc/blob/feed0fea2c5280264ac000947b3fee542d7d9e6a/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/TagBuilder.cs#L336